### PR TITLE
Minor header/api cleanup

### PR DESCRIPTION
**Pull request by nashif**
- kernel: move sched_priq.h to internal/ folder
- kernel: move z_is_thread_essential out of public kernel header
- kernel: move internal smp calls to a internal header
- kernel: remove unused z_init_thread_base from kernel.h
- kernel: move more internal smp calls into internal domain
- kernel: sys_clock: remove stray z_enable_sys_clock prototype
- sys_clock: header: minor cleanup and doxygenization

### CODE REVIEW

File: include/zephyr/sys_clock.h [hunk 1]
```diff
@@ -79,6 +79,36 @@ typedef struct {
  */
 #define K_TIMEOUT_EQ(a, b) ((a).ticks == (b).ticks)
 
+/** number of nsec per usec */
+#define NSEC_PER_USEC 1000U
+
+/** number of nsec per msec */
```
Comment (kartben) [inline include/zephyr/sys_clock.h:85]: for consistency you should spell the units out in these two comments

File: include/zephyr/sys_clock.h [hunk 2]
```diff
@@ -79,6 +79,36 @@ typedef struct {
  */
 #define K_TIMEOUT_EQ(a, b) ((a).ticks == (b).ticks)
 
+/** number of nsec per usec */
```
Comment (kartben) [inline include/zephyr/sys_clock.h:82]: maybe for a later update but these constants might be worth grouping in a `@name`d section. There are in fact probably several groups of such macros that could be grouped -- see below.
I know that you want to work on this incrementally but should you catch low-hanging fruits with regards to adding more structure where you can, please go for it :)

image

**Review by kartben — DISMISSED** [review]
Really only looked at actual docs and can't comment for the rest of the refactoring

Please also add comment for `K_TICKS_FOREVER` (L51 in sys_clock.h) and `SYS_CLOCK_HW_CYCLES_TO_NS_AVG` (L178, looks like it's just missing javadoc style comment) as it would be too bad on missing out on the opportunity to have this header fully documented :)

File: include/zephyr/sys_clock.h [hunk 3]
```diff
@@ -79,6 +79,36 @@ typedef struct {
...
+#define MIN_PER_HOUR 60U
+
+/** number of hours per day */
+#define HOUR_PER_DAY 24U
+
+/** number of microseconds per second */
+#define USEC_PER_SEC ((USEC_PER_MSEC) * (MSEC_PER_SEC))
+
+/** number of nanoseconds per second */
+#define NSEC_PER_SEC ((NSEC_PER_USEC) * (USEC_PER_MSEC) * (MSEC_PER_SEC))
```
Comment (kartben) [inline include/zephyr/sys_clock.h:107]: maybe for a later update but these time-conversion related constants might be worth grouping in a `@name`d section. There are in fact probably several groups of such macros that could be grouped -- see below.
I know that you want to work on this incrementally but should you catch low-hanging fruits with regards to adding more structure where you can, please go for it :)

image

Comment (nashif) [reply include/zephyr/sys_clock.h]: yeah, will do in a second round, right now I want to focus on removing private/internal mostly and low hanging fruit, ie.. misconfigured doxygen comments, not add comments myself and go into loops of reviews about documentation and content that I might get wrong.

**Review by peter-mitsis — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**